torchbenchmark/score/compute_score_v1.py

```diff
@@ -0,0 +1,193 @@
+        """
+        if ref_data == TORCHBENCH_V1_REF_DATA:
+            with open(ref_data) as ref_file:
+                ref = yaml.full_load(ref_file)
+        else:
+            ref = self._get_norm_from_ref_data(ref_data)
+        return ref
+
+    def _get_norm_from_ref_data(self, ref_data):
+        """
+        This function iterates over the reference data (json object)
+        and calculates the normalization values based on the reference data.
+        It also sets up the domain weights of the score.
+        """
+        norm = {}
+        norm['benchmarks'] = {}
+        for b in ref_data['benchmarks']:
+            d = {}
+            d['norm'] = b['stats']['mean']
+            norm['benchmarks'][b['name']] = d
+        return norm
+
+    def _get_score(self, data, ref, ref_weights):
+        score = 0.0
+        (domain_weights, config_weights) = ref_weights
+        for name in data['benchmarks']:
+            norm = data['benchmarks'][name]['norm']
+            benchmark_score = domain_weights[name] * config_weights[name] * math.log(norm / ref['benchmarks'][name]['norm'])
+            score += benchmark_score
+        return math.exp(score)
+
+    def data_in_list(self, n, l):
+        for e in l:
+            if e not in n:
+                return False
+        return True
+
+    def _get_subscore(self, data, ref_norm, ref_weights, filters):
+        assert len(filters) == 2, error_msg
+        assert "cpu" in filters or "cuda" in filters, error_msg
+        assert "train" in filters or "eval" in filters, error_msg
+        score = 0.0
+        (domain_weights, _) = ref_weights
+        for name in filter(lambda x: self.data_in_list(x, filters), data['benchmarks']):
+            norm = data['benchmarks'][name]['norm']
+            benchmark_score = domain_weights[name] * math.log(norm / ref_norm['benchmarks'][name]['norm'])
+            score += benchmark_score
+        return math.exp(score)
+    
+    def compute_jit_speedup_score(self, data):
+        """
+        This API calculates the V1 JIT speedup score for all 
+        the benchmarks that enable JIT compilation.
+        The data argument is the json data object from the benchmark.
+        The JIT speedup score is the geometric mean of all JIT benchmarks speedup
+        comparing to corresponding non-JIT benchmarks.
+        The weights are then calibrated to the target score.
+        """
+        score = 0.0
+        norm = self._setup_benchmark_norms(data)
+        norm_jit = self._filter_jit_tests(norm)
+        (domain_weights, config_weights) = self._setup_weights(norm_jit)
+        for name in norm_jit['benchmarks']:
+            eager_norm = norm_jit['benchmarks'][name]['eager_norm']
+            jit_norm = norm_jit['benchmarks'][name]['jit_norm']
+            jit_speedup_score = domain_weights[name] * config_weights[name] * math.log(eager_norm / jit_norm)
+            score += jit_speedup_score
+        return math.exp(score) * self.target
```

Comment:
does it make sense to use the target for the JIT speedup score?  I think if you're making the speedup score a special 'relative' computation, you can just leave it normalized to 1, so you'd get explainable numbers like 1.2x speedup or something.

Reply:
Sounds good to me. JIT speedup score is an absolute score and doesn't rely on reference data, so it can be normalized to 1.